pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentUploadDownloadRestletResource.java

```diff
@@ -292,12 +292,12 @@ private SuccessResponse uploadSegment(@Nullable String tableName, TableType tabl
                 tableNameWithType);
         zkDownloadUri = downloadUri;
       } else {
-        zkDownloadUri = getZkDownloadURIForSegmentUpload(tableNameWithType, segmentName);
+        zkDownloadUri = getZkDownloadURIForSegmentUpload(tableName, segmentName);
```

Comment:
It might be good to rename `tableName` to `rawTableName` given the parameter of `completeZkOperations()` is rawTableName?

Reply:
Good that you pointed that out. There's an execution path of this function where tableName might be null or empty and rawTableName is taken from metadata. So used rawTableName that's available.